src/graph.py

```diff
@@ -716,14 +758,22 @@ def get_robust_subgraph_from_randomizations(nxgraph, max_size=400, min_component
 
     # Get indices of top nodes sorted by high robustness, then low specificity. Don't allow nodes with robustness = 0.
     node_attributes_df = get_networkx_graph_as_dataframe_of_nodes(nxgraph)
-    node_attributes_df = node_attributes_df[node_attributes_df["robustness"] > 0]
+    node_attributes_df = node_attributes_df[node_attributes_df["robustness"] > min_robustness]
     node_attributes_df.sort_values(["robustness", "specificity"], ascending=[False, True], inplace=True)
-    top_hits = node_attributes_df.index[:min(max_size,len(node_attributes_df))]
-    # Get robust subnetwork and remove small components.
-    robust_network = nxgraph.subgraph(top_hits)
-    robust_network = filter_graph_by_component_size(robust_network, min_component_size)
 
-    return robust_network
+    # Find the largest subgraph of `nxgraph` such that the number of nodes, after filtering out small components, is 
+    # not more than `max_size`. This is accomplished by progressively loosening the threshold for inclusion in the subgraph. 
+    for top_n in np.arange(min(max_size, len(node_attributes_df)), len(node_attributes_df)+1): 
+        # Get robust subnetwork and remove small components.
+        robust_network = nxgraph.subgraph(node_attributes_df.index[:top_n])
+        robust_network = filter_graph_by_component_size(robust_network, min_component_size)
+
+        if robust_network.number_of_nodes() <= max_size: 
+            robust_network_out = robust_network
+        else: 
+            return robust_network_out
```

Comment:
I'm confused about this else block. What is `robust_network_out` when you go through the else?

Reply:
It's an early return statement so the entire for loop does not need to be executed. Once a robust network is found with more nodes than max size, no need to check less stringent thresholds.